### DISCUSSION TITLE
I'm looking for an opportunity to add a validator for tokens

### DISCUSSION THREAD

**Discussion by ihumster** [netbox-community/netbox#18189] [Help Wanted!] [answered]
For security reasons, it is necessary to prohibit the creation of tokens without expiration or with a validity period of more than 1 year (for example).

I tried to go the route of creating a Custom Validator (to create Custom Logics), but unfortunately Netbox does not connect this validator to `users.token`.

Is there such a possibility?

**Comment by jeremystretch** [comment, marked as answer]
Custom validators aren't currently supported for tokens, as you note, because the Token model doesn't inherit `CustomValidationMixin`, which provides this functionality. However I see no reason it couldn't be added. You're welcome to submit a [feature request](https://github.com/netbox-community/netbox/issues/new?assignees=&labels=type%3A+feature%2Cstatus%3A+needs+triage&projects=&template=01-feature_request.yaml) citing your use case.

That said, I'd also entertain a proposal to introduce a new configuration parameter asserting a minimum expiration time for API tokens. I imagine it's a fairly common need.
